src/Datadog.Trace/DiagnosticListeners/DiagnosticObserver.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.DiagnosticListeners
+{
+    internal abstract class DiagnosticObserver : IObserver<KeyValuePair<string, object>>
+    {
+        private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<DiagnosticObserver>();
+
+        protected DiagnosticObserver(IDatadogTracer tracer)
+        {
+            Tracer = tracer ?? throw new ArgumentNullException(nameof(tracer));
+        }
+
+        protected IDatadogTracer Tracer { get; }
+
+        /// <summary>
+        /// Gets the name of the <see cref="DiagnosticListener"/> that should be instrumented.
+        /// </summary>
+        /// <value>The name of the <see cref="DiagnosticListener"/> that should be instrumented.</value>
+        protected abstract string ListenerName { get; }
+
+        public virtual bool IsSubscriberEnabled()
+        {
+            return true;
```

Comment:
Any plans to make this config based?

Reply:
We should! This is left-over from the project I based these changes on. I can remove it or we can leave it as a placeholder for "future feature". I have no strong feelings either way.